src/components/app/index.tsx

```diff
@@ -133,57 +156,53 @@ export default class App extends Component<Props, State> {
   }
 
   async updateImage(index: number): Promise<void> {
-    const { source, images } = this.state;
+    const { source } = this.state;
     if (!source) return;
-    let image = images[index];
+    let images = this.state.images.slice() as [EncodedImage, EncodedImage];
 
-    // Each time we trigger an async encode, the ID changes.
-    image.loadingCounter = image.loadingCounter + 1;
-    const loadingCounter = image.loadingCounter;
+    // Each time we trigger an async encode, the counter changes.
+    const loadingCounter = images[index].loadingCounter + 1;
 
-    image.loading = true;
-    this.setState({ });
-
-    const result = await this.updateCompressedImage(source, image.encoderState);
+    const image = images[index] = {
+      ...images[index],
+      loadingCounter,
+      loading: true,
+    };
 
-    image = this.state.images[index];
-    // If a later encode has landed before this one, return.
-    if (loadingCounter < image.loadedCounter) return;
-    image.bmp = result;
-    image.loading = image.loadingCounter !== loadingCounter;
-    image.loadedCounter = loadingCounter;
-    this.setState({ });
-  }
+    this.setState({ images });
```

Comment:
I'm trying to avoid editing state & instead setting new state.

Reply:
Was going to suggest that in the other PR, glad its added here.